test/rest-api/generator.mocks.ts

```diff
@@ -0,0 +1,37 @@
+import { setupWorker, rest } from 'msw'
+
+const worker = setupWorker(
+  rest.get(
+    'https://test.mswjs.io/polling-endpoint/:count',
+    function* (req, res, ctx) {
+      const { count: maxCount } = req.params
+      let count = 0
+
+      while (count < maxCount) {
+        count += 1
+        yield res(
+          ctx.json({
+            status: 'pending',
+            count,
+          }),
+        )
+      }
+
+      return res(
```

Comment:
Would it be a common practice to keep returning this "complete" response once the generator is done?
It's not semantic to mark the handler as skipped, making it a one-time handler, without using the `res.once`. 

I suggest we achieve this use case by using `res.once`, and drop the `if (this.resolverGeneratorIsDone)` part from the request handling altogether. This way if you wish to respond once with the "complete" state, you'd express your intention explicitly:

```js
rest.get('/:count', function*(req, res, ctx) {
  while (predicate) {
    yield res(ctx.text('pending'))
  }

  // Once the generator is done, respond with the "complete"
  // state _once_. After this point this handler will self-destruct,
  // respecting the existing one-time handlers behavior.
  return res.once(ctx.text('complete'))
})
```

What do you think about this?

Reply:
Yeah, I think that's probably better. It means you can have the move-on-once-completed behaviour if you want it (with `res.once`), but allows you to also just have the "complete" response repeat if you wish.